### PR TITLE
feat(MyOnlineRadio): add presence

### PR DESCRIPTION
**Pull request by ION606**
## Description 
I believe I have completed the request found at https://github.com/PreMiD/Presences/issues/8078. I should note that while the presence will show which version of the website you are on (i.e. "France" if ".fr"), the only full support is for the UK/Ireland version, as they are in English.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/M/MyOnlineRadio/presence.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import { handleStation } from "./handleStation";
+
+const presence = new Presence({
+	clientId: "1217153856665026580",
+});
+
+// Other default assets can be found at index.d.ts
+const enum Assets {
+	Logo = "https://ukradiolive.com/public/img/launcher/logo_192.png",
+	smallImageKey = "",
```
Comment (Bas950) [inline websites/M/MyOnlineRadio/presence.ts:12]:
```suggestion
```

File: websites/M/MyOnlineRadio/presence.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import { getDropDownSelected, getInp, toDate } from "./utils";
+import { handleStation } from "./handleStation";
+
+const presence = new Presence({
+	clientId: "1217153856665026580",
+});
+
+// Other default assets can be found at index.d.ts
+const enum Assets {
+	Logo = "https://ukradiolive.com/public/img/launcher/logo_192.png",
```
Comment (Bas950) [inline websites/M/MyOnlineRadio/presence.ts:11]: Must be 512x512

File: websites/M/MyOnlineRadio/metadata.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		"myradioendirect.fr",
+		"irishradiolive.com",
+		"myradioonline.pl",
+		"myonlineradio.nl",
+		"myradioonline.cl",
+		"myradioenvivo.ar",
+		"myradioenvivo.mx"
+	],
+	"version": "1.0.0",
+	"logo": "https://ukradiolive.com/public/img/launcher/logo_192.png",
```
Comment (Bas950) [inline websites/M/MyOnlineRadio/metadata.json:29]: Logo must be 512x512

File: websites/M/MyOnlineRadio/handleStation.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+				presenceData.startTimestamp = tDiffInMS(
+					document.querySelector(".ctime").textContent
+				);
+				presenceData.largeImageText = document
+					.querySelector(".server-conts")
+					?.querySelector(".active > .info")?.textContent;
+
+				// check if they're recording
+				if (document.querySelector(".live")) {
+					presenceData.smallImageKey = "https://i.imgur.com/4xhphHG.png";
```
Comment (Bas950) [inline websites/M/MyOnlineRadio/handleStation.ts:129]: Image returning 404 not found, please try uploading to a different image website, imgur seems to be unstable recently.

**Review by Bas950 — CHANGES_REQUESTED** [review]
Remove the package-lock file changes.

**Comment by ION606** [comment]
oops my bad I think I ran npm update

File: websites/M/MyOnlineRadio/handleStation.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
+import { getDropDownSelected, getInp, toDate, tDiffInMS } from "./utils";
+
+/**
+ * @param station
+ */
```
Comment (theusaf) [inline websites/M/MyOnlineRadio/handleStation.ts:5]: This comment seems inaccurate or unnecessary

File: package-lock.json
Comment (theusaf) [inline package-lock.json:1]: Ensure to revert changes to this file

File: package.json
Comment (theusaf) [inline package.json:1]: Ditto

File: websites/M/MyOnlineRadio/handleStation.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+				if (document.querySelector(".live")) {
+					presenceData.smallImageKey = "https://i.ibb.co/QNhnC3z/mic.png";
+					presenceData.smallImageText = "Recording Broadcast";
+				}
+			}
+		}
+	}
+
+	presenceData.buttons = [
+		{ label: "View Station", url: `${window.location.origin}/${station}/` },
```
Comment (theusaf) [inline websites/M/MyOnlineRadio/handleStation.ts:137]:
```suggestion
		{ label: "View Station", url: `${document.location.origin}/${station}/` },
```

File: websites/M/MyOnlineRadio/metadata.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		"myradioonline.es",
+		"myradioendirect.fr",
+		"irishradiolive.com",
+		"myradioonline.pl",
+		"myonlineradio.nl",
+		"myradioonline.cl",
+		"myradioenvivo.ar",
+		"myradioenvivo.mx"
+	],
+	"version": "1.0.0",
```
Comment (theusaf) [inline websites/M/MyOnlineRadio/metadata.json:28]: There are new fields called `matches`, which are required for new Presences .
These are an array of strings which need to follow https://developer.chrome.com/docs/extensions/develop/concepts/match-patterns.

File: websites/M/MyOnlineRadio/presence.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
+import { extToCountry } from "./extToCountry";
+import { getDropDownSelected, getInp, toDate } from "./utils";
+import { handleStation } from "./handleStation";
+
+const presence = new Presence({
+	clientId: "1217153856665026580",
+});
+
+// Other default assets can be found at index.d.ts
```
Comment (theusaf) [inline websites/M/MyOnlineRadio/presence.ts:9]:
```suggestion
```

File: websites/M/MyOnlineRadio/handleStation.ts [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+				presenceData.startTimestamp = tDiffInMS(
+					document.querySelector(".ctime").textContent
+				);
+				presenceData.largeImageText = document
+					.querySelector(".server-conts")
+					?.querySelector(".active > .info")?.textContent;
+
+				// check if they're recording
+				if (document.querySelector(".live")) {
+					presenceData.smallImageKey = "https://i.ibb.co/QNhnC3z/mic.png";
```
Comment (theusaf) [inline websites/M/MyOnlineRadio/handleStation.ts:129]: Place this asset in the `Assets` enum

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/MyOnlineRadio/metadata.json (hunk 7 above)
Comment (ION606) [reply websites/M/MyOnlineRadio/metadata.json:28]: ohhhh so like `https://myonlineradio.hu/*`?

Comment (theusaf) [reply websites/M/MyOnlineRadio/metadata.json:28]: yeah

Comment (theusaf) [reply websites/M/MyOnlineRadio/metadata.json:28]: they don't replace the `url` section, but are a new section for future versions of PreMiD

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**